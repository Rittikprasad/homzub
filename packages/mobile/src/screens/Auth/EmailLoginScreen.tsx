import React from 'react';
import { StyleSheet } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import { IRedirectionDetails } from '@homzhub/mobile/src/services/LinkingService';
import { NavigationService } from '@homzhub/mobile/src/services/NavigationService';
import { UserActions } from '@homzhub/common/src/modules/user/actions';
import { CommonSelectors } from '@homzhub/common/src/modules/common/selectors';
import { UserSelector } from '@homzhub/common/src/modules/user/selectors';
import { theme } from '@homzhub/common/src/styles/theme';
import { icons } from '@homzhub/common/src/assets/icon';
import { LoginForm } from '@homzhub/common/src/components/organisms/LoginForm';
import { Screen } from '@homzhub/mobile/src/components/HOC/Screen';
import { AuthStackParamList } from '@homzhub/mobile/src/navigation/AuthStack';
import { IState } from '@homzhub/common/src/modules/interfaces';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import {
  IEmailLoginPayload,
  ILoginFormData,
  ILoginPayload,
  LoginTypes,
} from '@homzhub/common/src/domain/repositories/interfaces';

interface IStateProps {
  isLoading: boolean;
  redirectionDetails: IRedirectionDetails;
}

interface IDispatchProps {
  login: (payload: ILoginPayload) => void;
}

type libraryProps = WithTranslation & NavigationScreenProps<AuthStackParamList, ScreensKeys.EmailLogin>;
type Props = IDispatchProps & IStateProps & libraryProps;

export class EmailLoginScreen extends React.PureComponent<Props> {
  public render(): React.ReactNode {
    const { t, isLoading } = this.props;
    return (
      <Screen
        containerStyle={styles.container}
        headerProps={{
          type: 'secondary',
          icon: icons.leftArrow,
          onIconPress: this.handleIconPress,
        }}
        pageHeaderProps={{
          contentTitle: t('auth:logInWithEmail'),
        }}
        backgroundColor={theme.colors.white}
        keyboardShouldPersistTaps
        isLoading={isLoading}
      >
        <LoginForm
          isEmailLogin
          handleForgotPassword={this.handleForgotPassword}
          onLoginSuccess={this.handleLoginSuccess}
          testID="loginForm"
        />
      </Screen>
    );
  }

  private handleIconPress = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };

  private handleForgotPassword = (): void => {
    const {
      navigation,
      route: { params },
    } = this.props;
    const onCallback = params && params.onCallback ? { onCallback: params.onCallback } : {};
    navigation.navigate(ScreensKeys.ForgotPassword, onCallback);
  };

  private handleLoginSuccess = (values: ILoginFormData): void => {
    const {
      login,
      route: { params },
    } = this.props;

    const emailLoginData: IEmailLoginPayload = {
      action: LoginTypes.EMAIL,
      payload: {
        email: values.email,
        password: values.password,
      },
    };

    const loginPayload: ILoginPayload = {
      data: emailLoginData,
      ...(params && params.onCallback && { callback: params.onCallback }),
      handleDynamicLink: this.handleDynamicLink,
    };

    login(loginPayload);
  };

  private handleDynamicLink = (): void => {
    const { redirectionDetails } = this.props;
    if (redirectionDetails.shouldRedirect && redirectionDetails.redirectionLink) {
      NavigationService.handleDynamicLinkNavigation(redirectionDetails).then();
    }
  };
}

export const mapStateToProps = (state: IState): IStateProps => {
  return {
    isLoading: UserSelector.getLoadingState(state),
    redirectionDetails: CommonSelectors.getRedirectionDetails(state) as IRedirectionDetails,
  };
};

export const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  const { login } = UserActions;
  return bindActionCreators(
    {
      login,
    },
    dispatch
  );
};

// @ts-ignore
export default connect<IStateProps, IDispatchProps, WithTranslation, IState>(
  mapStateToProps,
  mapDispatchToProps
)(withTranslation()(EmailLoginScreen));

// @ts-ignore
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: theme.colors.white,
  },
});
