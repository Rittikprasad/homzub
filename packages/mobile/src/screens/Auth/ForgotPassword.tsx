import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { Formik, FormikProps, FormikValues } from 'formik';
import * as yup from 'yup';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { FormUtils } from '@homzhub/common/src/utils/FormUtils';
import { UserRepository } from '@homzhub/common/src/domain/repositories/UserRepository';
import { IForgotPasswordPayload } from '@homzhub/common/src/domain/repositories/interfaces';
import { theme } from '@homzhub/common/src/styles/theme';
import { icons } from '@homzhub/common/src/assets/icon';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { FormButton } from '@homzhub/common/src/components/molecules/FormButton';
import { FormTextInput } from '@homzhub/common/src/components/molecules/FormTextInput';
import { Screen } from '@homzhub/mobile/src/components/HOC/Screen';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { AuthStackParamList } from '@homzhub/mobile/src/navigation/AuthStack';

interface IForgotPasswordState {
  email: string;
  isRequestDisable: boolean;
}

type Props = WithTranslation & NavigationScreenProps<AuthStackParamList, ScreensKeys.ForgotPassword>;

export class ForgotPassword extends Component<Props, IForgotPasswordState> {
  public state = {
    email: '',
    isRequestDisable: false,
  };

  public render(): React.ReactNode {
    const {
      t,
      route: { params },
    } = this.props;
    const { isRequestDisable } = this.state;
    const formData = { ...this.state };
    return (
      <Screen
        headerProps={{
          type: 'secondary',
          icon: icons.close,
          onIconPress: this.handleIconPress,
        }}
        pageHeaderProps={{
          contentTitle: t('auth:forgotPassword'),
          contentSubTitle: t('auth:forgotPasswordDescription'),
          disableDivider: true,
        }}
        backgroundColor={theme.colors.white}
      >
        <Formik onSubmit={this.onSubmit} validate={FormUtils.validate(this.formSchema)} initialValues={formData}>
          {(formProps: FormikProps<FormikValues>): React.ReactElement => (
            <>
              <FormTextInput
                formProps={formProps}
                inputType="email"
                name="email"
                label="Email"
                placeholder={t('auth:enterEmail')}
                isMandatory
                onValueChange={this.onValueChange}
              />
              <FormButton
                formProps={formProps}
                type="primary"
                title={t('auth:forgotPasswordRequestLink')}
                // @ts-ignore
                onPress={formProps.handleSubmit}
                containerStyle={styles.formButtonStyle}
                disabled={!formProps.values.email || (formProps.values.email && isRequestDisable)}
              />
            </>
          )}
        </Formik>
        {params && !params.isFromMore && (
          <Text
            type="small"
            textType="semiBold"
            style={styles.backToLoginLink}
            onPress={this.navigateToLogin}
            testID="txtLogin"
          >
            {t('auth:backToLogin')}
          </Text>
        )}
      </Screen>
    );
  }

  private onValueChange = (): void => {
    const { isRequestDisable } = this.state;
    if (isRequestDisable) {
      this.setState({ isRequestDisable: false });
    }
  };

  private onSubmit = async (formProps: IForgotPasswordState): Promise<void> => {
    const { t } = this.props;
    const { email } = formProps;
    const payload: IForgotPasswordPayload = {
      action: 'SEND_EMAIL',
      payload: {
        email,
      },
    };

    this.setState({ isRequestDisable: true });

    try {
      const emailData = await UserRepository.emailExists(email);
      if (emailData.is_exists) {
        UserRepository.resetPassword(payload)
          .then(() => {
            AlertHelper.success({ message: t('auth:resetLink') });
          })
          .catch((err) => {
            AlertHelper.error({ message: err });
          });
      } else {
        AlertHelper.error({ message: t('auth:emailNotExists') });
      }
    } catch (e) {
      AlertHelper.error({ message: e.message });
    }
  };

  public navigateToLogin = (): void => {
    const {
      navigation,
      route: { params },
    } = this.props;
    const onCallback = params && params.onCallback ? { onCallback: params.onCallback } : {};
    navigation.navigate(ScreensKeys.EmailLogin, onCallback);
  };

  public handleIconPress = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };

  private formSchema = (): yup.ObjectSchema<{ email: string }> => {
    const { t } = this.props;
    return yup.object().shape({
      email: yup.string().email(t('auth:emailValidation')).required(t('auth:emailRequired')),
    });
  };
}

export default withTranslation()(ForgotPassword);

const styles = StyleSheet.create({
  formButtonStyle: {
    flex: 0,
    marginVertical: 30,
  },
  backToLoginLink: {
    color: theme.colors.blue,
    alignSelf: 'center',
  },
});
