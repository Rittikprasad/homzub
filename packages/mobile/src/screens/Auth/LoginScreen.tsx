import React, { Component } from 'react';
import { withTranslation, WithTranslation } from 'react-i18next';
import { theme } from '@homzhub/common/src/styles/theme';
import { icons } from '@homzhub/common/src/assets/icon';
import { AuthenticationGateways } from '@homzhub/mobile/src/components';
import { LoginForm } from '@homzhub/common/src/components/organisms/LoginForm';
import { Screen } from '@homzhub/mobile/src/components/HOC/Screen';
import { NavigationScreenProps, OtpNavTypes, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { AuthStackParamList } from '@homzhub/mobile/src/navigation/AuthStack';
import { ILoginFormData } from '@homzhub/common/src/domain/repositories/interfaces';

type Props = WithTranslation & NavigationScreenProps<AuthStackParamList, ScreensKeys.Login>;

interface IOwnState {
  isLoading: boolean;
}

export class LoginScreen extends Component<Props, IOwnState> {
  public state = {
    isLoading: false,
  };

  public render(): React.ReactNode {
    const {
      t,
      navigation,
      route: { params },
    } = this.props;
    const { isLoading } = this.state;

    return (
      <Screen
        headerProps={{
          type: 'secondary',
          icon: icons.close,
          onIconPress: this.onClosePress,
        }}
        pageHeaderProps={{
          contentTitle: t('login'),
          contentSubTitle: t('auth:newAroundHere'),
          contentLink: t('auth:signup'),
          onLinkPress: this.onSignUpClicked,
        }}
        backgroundColor={theme.colors.white}
        isLoading={isLoading}
      >
        <>
          <LoginForm onLoginSuccess={this.onOtpLoginPress} testID="loginForm" />
          <AuthenticationGateways
            isFromLogin
            onEmailLogin={this.onEmailLoginPress}
            onSuccessCallback={params.onCallback}
            navigation={navigation}
            testID="socialEmailLogin"
            toggleLoading={this.toggleLoading}
          />
        </>
      </Screen>
    );
  }

  private onClosePress = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };

  private onEmailLoginPress = (): void => {
    const {
      navigation,
      route: { params },
    } = this.props;
    const onCallback = params && params.onCallback ? { onCallback: params.onCallback } : {};
    navigation.navigate(ScreensKeys.EmailLogin, onCallback);
  };

  private onSignUpClicked = (): void => {
    const {
      navigation,
      route: { params },
    } = this.props;
    const onCallback = params && params.onCallback ? { onCallback: params.onCallback } : {};
    navigation.navigate(ScreensKeys.SignUp, onCallback);
  };

  private onOtpLoginPress = (values: ILoginFormData): void => {
    const {
      navigation,
      t,
      route: { params },
    } = this.props;
    navigation.navigate(ScreensKeys.OTP, {
      type: OtpNavTypes.Login,
      title: t('auth:loginOtp') ?? '',
      countryCode: values.phone_code,
      otpSentTo: values.phone_number,
      ...(params && params.onCallback && { onCallback: params.onCallback }),
    });
  };

  private toggleLoading = (isLoading: boolean): void => {
    this.setState({ isLoading });
  };
}

export default withTranslation()(LoginScreen);
