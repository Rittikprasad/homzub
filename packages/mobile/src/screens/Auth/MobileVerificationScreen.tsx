import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import { Formik, FormikHelpers, FormikProps } from 'formik';
import * as yup from 'yup';
import { withTranslation, WithTranslation } from 'react-i18next';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { ErrorUtils } from '@homzhub/common/src/utils/ErrorUtils';
import { FormUtils } from '@homzhub/common/src/utils/FormUtils';
import { UserRepository } from '@homzhub/common/src/domain/repositories/UserRepository';
import { theme } from '@homzhub/common/src/styles/theme';
import { Text } from '@homzhub/common/src/components/atoms/Text';
import { FormButton } from '@homzhub/common/src/components/molecules/FormButton';
import { FormTextInput } from '@homzhub/common/src/components/molecules/FormTextInput';
import { Screen } from '@homzhub/mobile/src/components/HOC/Screen';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { AuthStackParamList } from '@homzhub/mobile/src/navigation/AuthStack';
import { NavigationScreenProps, OtpNavTypes, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { SocialAuthKeys } from '@homzhub/common/src/constants/SocialAuthProviders';

interface IVerificationState {
  phone: string;
  phoneCode: string;
}

type Props = WithTranslation & NavigationScreenProps<AuthStackParamList, ScreensKeys.MobileVerification>;

export class MobileVerificationScreen extends Component<Props, IVerificationState> {
  public state = {
    phone: '',
    phoneCode: '',
  };

  public render(): React.ReactNode {
    const { t } = this.props;
    const { title, message, subTitle, buttonTitle } = this.getDisplayStrings();

    return (
      <Screen
        headerProps={{
          type: 'secondary',
          onIconPress: this.handleIconPress,
        }}
        pageHeaderProps={{
          contentTitle: title,
          contentSubTitle: subTitle,
        }}
        backgroundColor={theme.colors.white}
        contentContainerStyle={styles.container}
      >
        <Text type="small" style={styles.message}>
          {message}
        </Text>
        <Formik
          initialValues={{ ...this.state }}
          onSubmit={this.onSubmit}
          validate={FormUtils.validate(this.formSchema)}
        >
          {(formProps: FormikProps<IVerificationState>): React.ReactElement => (
            <>
              <FormTextInput
                formProps={formProps}
                inputType="phone"
                name="phone"
                label="Phone"
                inputPrefixText={formProps.values.phoneCode}
                isMandatory
                phoneFieldDropdownText={t('auth:countryRegion')}
                placeholder={t('yourNumber')}
                helpText={t('otpVerification')}
              />
              <FormButton
                // @ts-ignore
                onPress={formProps.handleSubmit}
                formProps={formProps}
                type="primary"
                title={buttonTitle}
                containerStyle={styles.formButtonStyle}
              />
            </>
          )}
        </Formik>
      </Screen>
    );
  }

  private onSubmit = async (
    values: IVerificationState,
    formActions: FormikHelpers<IVerificationState>
  ): Promise<void> => {
    formActions.setSubmitting(true);
    const {
      t,
      route: {
        params: { isFromLogin, userData, onCallback },
      },
      navigation,
    } = this.props;
    const { phone, phoneCode } = values;

    try {
      const formattedPhone = `${phoneCode}~${phone}`;
      const isPhoneUsed = await UserRepository.phoneExists(formattedPhone);
      if (isPhoneUsed.is_exists) {
        AlertHelper.error({ message: t('auth:phoneAlreadyExists') });
        return;
      }
      navigation.navigate(ScreensKeys.OTP, {
        type: OtpNavTypes.SocialMedia,
        title: isFromLogin ? t('loginOtp') : t('verifyNumber'),
        otpSentTo: phone,
        countryCode: phoneCode,
        socialUserData: userData,
        ...(onCallback && { onCallback }),
      });
    } catch (err) {
      AlertHelper.error({ message: ErrorUtils.getErrorMessage(err.details) });
    }
  };

  private handleIconPress = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };

  private getDisplayStrings = (): { title: string; subTitle: string; message: string; buttonTitle: string } => {
    const {
      t,
      route: {
        params: { userData, isFromLogin },
      },
    } = this.props;
    const {
      provider,
      user: { first_name, email },
    } = userData;

    let title = t('signUpWithFacebook');
    const messageKey = isFromLogin ? 'enterNumberForProfileForLogin' : 'enterNumberForProfileForSignUp';

    if (isFromLogin) {
      if (provider === SocialAuthKeys.Google) {
        title = t('loginWithGoogle');
      } else if (provider === SocialAuthKeys.Apple) {
        title = t('loginWithApple');
      } else {
        title = t('loginWithFacebook');
      }
    }

    if (!isFromLogin) {
      switch (provider) {
        case SocialAuthKeys.Google:
          title = t('signUpWithGoogle');
          break;
        case SocialAuthKeys.Apple:
          title = t('signUpWithApple');
          break;
        default:
          title = t('signUpWithFacebook');
      }
    }

    return {
      title,
      subTitle: email,
      buttonTitle: isFromLogin ? t('common:login') : t('common:signUp'),
      message: t(`${messageKey}`, { givenName: first_name }),
    };
  };

  private formSchema = (): yup.ObjectSchema<{ phone: string }> => {
    const { t } = this.props;
    return yup.object().shape({
      phone: yup.string().required(t('numberRequired')),
    });
  };
}

export default withTranslation(LocaleConstants.namespacesKey.auth)(MobileVerificationScreen);

const styles = StyleSheet.create({
  message: {
    color: theme.colors.darkTint3,
    marginBottom: 12,
  },
  formButtonStyle: {
    flex: 0,
    marginVertical: 30,
  },
  container: {
    paddingVertical: theme.layout.screenPadding,
  },
});
