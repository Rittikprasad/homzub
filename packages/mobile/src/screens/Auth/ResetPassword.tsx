import React, { Component } from 'react';
import { StyleSheet } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { Formik, FormikProps, FormikValues } from 'formik';
import * as yup from 'yup';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { ErrorUtils } from '@homzhub/common/src/utils/ErrorUtils';
import { FormUtils } from '@homzhub/common/src/utils/FormUtils';
import { UserRepository } from '@homzhub/common/src/domain/repositories/UserRepository';
import { IForgotPasswordPayload } from '@homzhub/common/src/domain/repositories/interfaces';
import { StoreProviderService } from '@homzhub/common/src/services/StoreProviderService';
import { IUserTokens, StorageKeys, StorageService } from '@homzhub/common/src/services/storage/StorageService';
import { theme } from '@homzhub/common/src/styles/theme';
import { icons } from '@homzhub/common/src/assets/icon';
import { FormButton } from '@homzhub/common/src/components/molecules/FormButton';
import { FormTextInput } from '@homzhub/common/src/components/molecules/FormTextInput';
import { Screen } from '@homzhub/mobile/src/components/HOC/Screen';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { AuthStackParamList } from '@homzhub/mobile/src/navigation/AuthStack';

type Props = WithTranslation & NavigationScreenProps<AuthStackParamList, ScreensKeys.ResetPassword>;

interface IResetPasswordState {
  password: string;
}

export class ResetPassword extends Component<Props, IResetPasswordState> {
  public state = {
    password: '',
  };

  public render(): React.ReactNode {
    const { t } = this.props;
    const formData = { ...this.state };
    return (
      <Screen
        headerProps={{
          type: 'secondary',
          icon: icons.close,
          onIconPress: this.handleIconPress,
        }}
        pageHeaderProps={{
          contentTitle: t('auth:resetPassword'),
          disableDivider: true,
        }}
        backgroundColor={theme.colors.white}
      >
        <Formik onSubmit={this.onSubmit} validate={FormUtils.validate(this.formSchema)} initialValues={formData}>
          {(formProps: FormikProps<FormikValues>): React.ReactElement => (
            <>
              <FormTextInput
                formProps={formProps}
                inputType="password"
                name="password"
                label="Password"
                helpText={t('auth:passwordValidation')}
                placeholder={t('auth:enterPassword')}
                isMandatory
              />
              <FormButton
                formProps={formProps}
                type="primary"
                title={t('auth:resetPasswordButtonTitle')}
                // @ts-ignore
                onPress={formProps.handleSubmit}
                containerStyle={styles.formButtonStyle}
              />
            </>
          )}
        </Formik>
      </Screen>
    );
  }

  private onSubmit = async (formProps: IResetPasswordState): Promise<void> => {
    const { password } = formProps;
    const userData = await StorageService.get<IUserTokens>(StorageKeys.USER);
    const {
      navigation,
      route: {
        params: { verification_id, invite_id },
      },
      t,
    } = this.props;
    const payload: IForgotPasswordPayload = {
      action: 'SET_PASSWORD',
      payload: {
        verification_id,
        invite_id,
        password,
      },
    };

    UserRepository.resetPassword(payload)
      .then(() => {
        if (userData) {
          StoreProviderService.logoutUserAndClearTokens();
          AlertHelper.success({ message: t('auth:PasswordChangedText') });
          return;
        }

        navigation.navigate(ScreensKeys.SuccessResetPassword);
      })
      .catch((e) => {
        AlertHelper.error({ message: ErrorUtils.getErrorMessage(e.details) });
      });
  };

  public handleIconPress = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };

  private formSchema = (): yup.ObjectSchema<{ password: string }> => {
    const { t } = this.props;
    return yup.object().shape({
      password: yup
        .string()
        .matches(FormUtils.passwordRegex, t('auth:passwordValidation'))
        .min(6, t('auth:minimumCharacters'))
        .required(t('auth:passwordRequired')),
    });
  };
}

export default withTranslation()(ResetPassword);

const styles = StyleSheet.create({
  formButtonStyle: {
    flex: 0,
    marginVertical: 30,
  },
});
