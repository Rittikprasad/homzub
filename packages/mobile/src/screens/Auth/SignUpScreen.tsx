import React, { Component } from 'react';
import { withTranslation, WithTranslation } from 'react-i18next';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { ErrorUtils } from '@homzhub/common/src/utils/ErrorUtils';
import { PlatformUtils } from '@homzhub/common/src/utils/PlatformUtils';
import { UserRepository } from '@homzhub/common/src/domain/repositories/UserRepository';
import { theme } from '@homzhub/common/src/styles/theme';
import { icons } from '@homzhub/common/src/assets/icon';
import { AuthenticationGateways } from '@homzhub/mobile/src/components';
import { SignUpForm } from '@homzhub/common/src/components/organisms/SignUpForm';
import { Screen } from '@homzhub/mobile/src/components/HOC/Screen';
import { ISignUpPayload } from '@homzhub/common/src/domain/repositories/interfaces';
import { AuthStackParamList } from '@homzhub/mobile/src/navigation/AuthStack';
import { NavigationScreenProps, OtpNavTypes, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';

type Props = WithTranslation & NavigationScreenProps<AuthStackParamList, ScreensKeys.SignUp>;

interface IOwnState {
  isLoading: boolean;
}

export class SignUpScreen extends Component<Props, IOwnState> {
  public state = {
    isLoading: false,
  };

  public render(): React.ReactNode {
    const {
      t,
      navigation,
      route: { params },
    } = this.props;
    const { isLoading } = this.state;

    return (
      <Screen
        headerProps={{
          type: 'secondary',
          icon: icons.close,
          onIconPress: this.onClosePress,
        }}
        pageHeaderProps={{
          contentTitle: t('signUp'),
          contentSubTitle: t('auth:alreadyRegistered'),
          contentLink: t('login'),
          onLinkPress: this.onLoginPress,
        }}
        backgroundColor={theme.colors.white}
        keyboardShouldPersistTaps
        isLoading={isLoading}
      >
        <>
          <SignUpForm
            onSubmitFormSuccess={this.onFormSubmit}
            onPressLink={this.handleTermsCondition}
            onPressPrivacyLink={this.handlePrivacy}
            referralCode={params && params?.referralCode}
            testID="signupForm"
          />
          {!PlatformUtils.isIOS() && (
            <AuthenticationGateways
              onSuccessCallback={params?.onCallback}
              toggleLoading={this.toggleLoading}
              isFromLogin={false}
              navigation={navigation}
            />
          )}
        </>
      </Screen>
    );
  }

  private onClosePress = (): void => {
    const { navigation } = this.props;
    navigation.goBack();
  };

  private onLoginPress = (): void => {
    const {
      navigation,
      route: { params },
    } = this.props;
    const onCallback = params && params.onCallback ? { onCallback: params.onCallback } : {};
    navigation.navigate(ScreensKeys.Login, onCallback);
  };

  private onFormSubmit = async (formData: ISignUpPayload): Promise<void> => {
    const {
      navigation,
      t,
      route: { params },
    } = this.props;

    try {
      const isEmailUsed = await UserRepository.emailExists(formData.email);
      if (isEmailUsed.is_exists) {
        AlertHelper.error({ message: t('auth:emailAlreadyExists') });
        return;
      }

      const phone = `${formData.phone_code}~${formData.phone_number}`;
      const isPhoneUsed = await UserRepository.phoneExists(phone);
      if (isPhoneUsed.is_exists) {
        AlertHelper.error({ message: t('auth:phoneAlreadyExists') });
        return;
      }

      if (formData.signup_referral_code) {
        const isValidCode = await UserRepository.verifyReferalCode(formData.signup_referral_code);
        if (!isValidCode.is_applicable) {
          AlertHelper.error({ message: t('auth:invalidReferralCodeError') });
          return;
        }
      }

      navigation.navigate(ScreensKeys.OTP, {
        type: OtpNavTypes.SignUp,
        title: t('auth:verifyNumber') ?? '',
        countryCode: formData.phone_code,
        otpSentTo: formData.phone_number,
        userData: formData,
        ...(params && params.onCallback && { onCallback: params.onCallback }),
      });
    } catch (err) {
      console.log('ðŸš€ ~ file: SignUpScreen.tsx ~ line 126 ~ SignUpScreen ~ onFormSubmit= ~ err.details', err.details);
      AlertHelper.error({ message: ErrorUtils.getErrorMessage(err.details) });
    }
  };

  private handleTermsCondition = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.WebViewScreen, { url: 'https://www.homzhub.com/terms&Condition' });
  };

  private handlePrivacy = (): void => {
    const { navigation } = this.props;
    navigation.navigate(ScreensKeys.WebViewScreen, { url: 'https://www.homzhub.com/privacyPolicy' });
  };

  private toggleLoading = (isLoading: boolean): void => {
    this.setState({ isLoading });
  };
}

export default withTranslation()(SignUpScreen);
