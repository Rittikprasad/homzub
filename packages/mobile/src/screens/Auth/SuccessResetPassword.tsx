import React, { Component } from 'react';
import { CommonActions } from '@react-navigation/native';
import { StyleSheet } from 'react-native';
import { withTranslation, WithTranslation } from 'react-i18next';
import { theme } from '@homzhub/common/src/styles/theme';
import { icons } from '@homzhub/common/src/assets/icon';
import { Button } from '@homzhub/common/src/components/atoms/Button';
import { Screen } from '@homzhub/mobile/src/components/HOC/Screen';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { CommonParamList } from '@homzhub/mobile/src/navigation/Common';

export type Props = WithTranslation & NavigationScreenProps<CommonParamList, ScreensKeys.SuccessResetPassword>;

export class SuccessResetPassword extends Component<Props, {}> {
  public render(): React.ReactNode {
    const { t } = this.props;
    return (
      <Screen
        headerProps={{
          type: 'secondary',
          icon: icons.close,
          onIconPress: this.navigateToLogin,
        }}
        pageHeaderProps={{
          contentTitle: t('auth:successResetPassword'),
          contentSubTitle: t('auth:successResetPasswordDescription'),
          disableDivider: true,
        }}
        backgroundColor={theme.colors.white}
      >
        <Button
          type="primary"
          title={t('common:login')}
          onPress={this.navigateToLogin}
          containerStyle={styles.button}
          testID="btnLogin"
        />
      </Screen>
    );
  }

  public navigateToLogin = (): void => {
    const { navigation } = this.props;

    navigation.dispatch(
      CommonActions.reset({
        index: 0,
        routes: [{ name: ScreensKeys.Login }],
      })
    );
  };
}

export default withTranslation()(SuccessResetPassword);

const styles = StyleSheet.create({
  button: {
    flex: 0,
  },
});
