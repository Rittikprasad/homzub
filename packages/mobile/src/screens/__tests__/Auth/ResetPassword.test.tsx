import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { ResetPassword } from '@homzhub/mobile/src/screens/Auth/ResetPassword';

const mock = jest.fn();

describe('Reset Password Screen', () => {
  let component: any;
  let props: any;

  beforeEach(() => {
    props = {
      navigation: {
        navigate: mock,
      },
    };
    component = shallow(
      <ResetPassword
        {...props}
        t={(key: string): string => key}
        route={{
          params: {
            onCallback: mock,
          },
        }}
      />
    );
  });

  it('should render reset password screen', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it.skip('should navigate on icon press', () => {
    component.find('[testID="headerIconPress"]').prop('onIconPress')();
    expect(mock).toHaveBeenCalled();
  });
});
