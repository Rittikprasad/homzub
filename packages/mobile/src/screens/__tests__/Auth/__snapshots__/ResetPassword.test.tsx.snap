// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reset Password Screen should render reset password screen 1`] = `
<Screen
  backgroundColor="#FFFFFF"
  headerProps={
    Object {
      "icon": "close",
      "onIconPress": [Function],
      "type": "secondary",
    }
  }
  pageHeaderProps={
    Object {
      "contentTitle": "auth:resetPassword",
      "disableDivider": true,
    }
  }
>
  <Formik
    initialValues={
      Object {
        "password": "",
      }
    }
    onSubmit={[Function]}
    validate={[Function]}
  >
    <Component />
  </Formik>
</Screen>
`;
